src/app/pages/SearchBar.tsx

```diff
@@ -0,0 +1,214 @@
+            flex: 1,
+        },
+        backdrop: {
+            zIndex: 999,
+        },
+        searchResultsOverlay: {
+            position: 'fixed',
+            marginTop: theme.spacing(8),
+            top: 0,
+            left: 0,
+            width: '100%',
+            height: '100%',
+            overflowY: 'auto',
+            zIndex: 1001,
+            backgroundColor: theme.palette.background.default,
+            [theme.breakpoints.down('xs')]: {
+                marginTop: theme.spacing(7),
+            },
+        },
+        searchResultsContainer: {
+            maxWidth: 644,
+            margin: '0 auto',
+            padding: `${theme.spacing(3)}px ${PADDING}px 20vh ${PADDING}px`,
+        },
+        searchResultCount: {
+            fontWeight: 600,
+            color: theme.palette.text.hint,
+        },
+        searchResult: {
+            '&:hover': {
+                cursor: 'pointer',
+                color: theme.palette.text.secondary,
+            },
+            '& p, & h6': {
+                marginBottom: theme.spacing(1),
+            },
+        },
+        searchResultPath: {
+            color: theme.palette.text.hint,
+        },
+        searchResultDivider: {
+            margin: `${theme.spacing(3)}px 0`,
+        },
+    })
+);
+
+export const SearchBar: React.FC<SearchbarProps> = (props) => {
+    const classes = useStyles();
+    const onSearch = useSelector((state: AppState) => state.app.onSearch);
+    const dispatch = useDispatch();
+    const [query, setQuery] = useState('');
+    const [searchResults, setSearchResults] = useState([]);
+    const [showSearchResult, setShowSearchResult] = useState(false);
+    const [inputString, setInputString] = useState('');
+    const history = useHistory();
+
+    React.useEffect(() => {
+        if (query) {
+            const loadSearchResults = async (): Promise<void> => {
+                const siteMapDB = await getSitemapDatabase();
+                const indexDB = await getIndexDatabase();
+                if (!siteMapDB || !indexDB) return;
+                // TODO
+                // @ts-ignore
+                setSearchResults(search(query, siteMapDB, indexDB));
+                setShowSearchResult(true);
+            };
+            loadSearchResults();
+        } else {
+            setShowSearchResult(false);
+        }
+    }, [query]);
+
+    // do auto suggestion stuff here
+    const onChangeHandler = (q: string): void => {
+        setInputString(q);
+        if (!q) {
+            setShowSearchResult(false);
+        }
+    };
+
+    const dismissSearchBar = (): void => {
+        setShowSearchResult(false);
+        setQuery('');
+        setInputString('');
+        setSearchResults([]);
+        dispatch({ type: TOGGLE_SEARCH, payload: false });
+    };
+
+    const getSearchResultCountText = (): string => {
+        switch (searchResults.length) {
+            case 0:
+                return `No result found for "${query}".`;
```

Comment:
```suggestion
                return `No results found for "${query}".`;
```

Reply:
It always perplexed me if zero counts as a plural or singular...